### PR TITLE
transformers: improvements to finetuning, run file, and bert 1mi results

### PR DESCRIPTION
**Pull request by lucasosouza**
This PR closes task1. Results are comparable to BERT paper. Updated BERT results, including 1mi:

|            | average_bert     | average_glue      | cola           | mnli                         | mrpc        | qnli     | qqp         | rte      | sst2     | stsb                  | wnli     |
|:-----------|:-----------------|:------------------|:---------------|:-----------------------------|:------------|:---------|:------------|:---------|:---------|:----------------------|:---------|
|            | Average w/o wnli | Average all tasks | Matthew's corr | Matched acc./Mismatched acc. | F1/Accuracy | Accuracy | Accuracy/F1 | Accuracy | Accuracy | Person/Spearman corr. | Accuracy |
| bert_HF    | 81.67            | 78.85             | 56.53          | 83.91/84.10                  | 88.85/84.07 | 90.66    | 90.71/87.49 | 65.70    | 92.32    | 88.64/88.48           | 56.34    |
| bert_paper|          79.60 |             -  |  52.10 | 84.60/83.40          | 88.90/- | 90.50     | 71.20/-     | 66.40     | 93.50     | 85.80        |      - |
| bert_1mi |          80.76 |          76.82 |  48.87 | 84.08/84.57 | 89.76/85.68 |  91.19 | 90.58/87.17 | 66.02 |  91.44 | 87.67/87.54 |  45.31 |
| bert_700k |          79.77 |          75.25 |  47.73 | 83.91/84.22 | 88.27/83.59 |  91.51 | 90.37/86.90 | 62.11 |  91.44 | 86.88/86.70 |  39.06 |
| bert_100k |          75.71 |          72.51 |  40.98 | 78.26/78.65 | 84.05/77.86 |  87.74 | 89.12/85.25 |  58.2 |  88.31 | 83.90/83.80 |  46.88 |

Results for the 700k checkpoint results are close to the 1mi checkpoint, and the 100k run is not so far off. These results should not be taken at face value - the same 1mi model finetuned in a previous run had significantly different results (+- 5 points in acc) even in tasks with large datasets such as MNLI. To effectively use GLUE results to compare models, we should improve our finetuning regime to include some form of hyperparameter optimization. 

Some pending issues:
- Finetuning regime needs improvement. I implemented averaging or maxing over several runs. But ideally we would like not only to max over several runs, but to also optimize hyperparameters either with grid search or using some method like hyperband or bayesian optimization. Grid search would be easy to implement with the current code structure, but better hyperparameter optimization methods for finetuning tasks might require some refactoring. 
- Missing sparse models. To be added after @mvacaporale PR on sparse bert models 

Changes worth mentioning:
- run.py no longer supports json and extra command line arguments, only experiments. Experiment is now a positional required argument and it accepts either a single experiment or a list of experiments to run sequentially - all other model, data or training arguments should be defined in config, and will continue to be parsed as before by a specific parser defined in run_args.py. The reasoning is: 1) if using list of experiments, passing args through command line no longer make sense  2) we have complex arguments, like trainer callbacks, that require instantiating a class 3) forces all arguments to be explicitly defined in config, making reproduction easier. `transformers-cli-utils` has also been modified to reflect that, please pull from master in infrastructure.
- `run_name` and `output_dir` are longer required arguments - default is to create a run folder based on experiment_name. This avoids a situation where the user creates a config based on another existing one but forgets to update the run_name, overriding a previously existing run. Default behavior is still to attempt to load existing checkpoint from a previous run with the same name, instead of initiating new run from scratch.

### CODE REVIEW

**Comment by lucasosouza** [comment]
@mvacaporale once you run sparse, can you please finetune using the same specs from finetuning_bert1mi_glue and post the results on the Readme. Thanks

File: projects/transformers/README.md [hunk 1]
```diff
@@ -33,7 +33,11 @@ Known issues, to be investigated/fixed:
...
 Local implementation using Huggingface. To run, create a new experiment dict under experiments and run using:
 
-`python run.py -e <experiment_name>`
+`python run.py <experiment_name>`
+
+You can also run multiple experiments in sequence:
+
+`python run.py <experiment_name_A>  <experiment_name_B>`
 
 Other accepted ways are passing arguments directly in command line or through a json file. See run.py for more details.
```
Comment (mvacaporale) [inline projects/transformers/README.md:42]: Didn't you remove this feature?

**Review by mvacaporale — APPROVED** [review]
Looks good. Just one small comment.

File: projects/transformers/README.md (hunk 1 above)
Comment (lucasosouza) [reply projects/transformers/README.md:42]: Good point

**Review by subutai — APPROVED** [review]
